pkg/watermark/fetch/edge_fetcher_set.go

```diff
@@ -59,6 +59,11 @@ func NewEdgeFetcherSet(ctx context.Context, vertexInstance *dfv1.VertexInstance,
 	}
 }
 
+func (efs *edgeFetcherSet) ComputeHeadWatermark(fromPartitionIdx int32) wmb.Watermark {
+	// won't be used
+	return wmb.Watermark{}
+}
```

Comment:
to avoid this, lets introduce you separate interface for source fetcher 

```
// SourceFetcher fetches watermark data for source vertex.
type SourceFetcher interface {
	// ComputeWatermark computes a valid watermark for the given offset on the given partition
	ComputeWatermark(offset isb.Offset, fromPartitionIdx int32) wmb.Watermark
	// ComputeHeadIdleWMB computes a valid head idle WMB for the given partition.
	ComputeHeadIdleWMB(fromPartitionIdx int32) wmb.WMB
	// ComputeHeadWatermark computes a valid head watermark for the given partition
	ComputeHeadWatermark(fromPartitionIdx int32) wmb.Watermark
}
```